instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/job/TracingJobExecutionListener.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.spring.batch.job;
+
+import static io.opentelemetry.javaagent.instrumentation.spring.batch.job.JobExecutionTracer.tracer;
+
+import io.opentelemetry.context.Context;
+import io.opentelemetry.context.Scope;
+import io.opentelemetry.javaagent.instrumentation.api.ContextStore;
+import io.opentelemetry.javaagent.instrumentation.spring.batch.ContextAndScope;
+import org.springframework.batch.core.JobExecution;
+import org.springframework.batch.core.JobExecutionListener;
+import org.springframework.core.Ordered;
+
+public class TracingJobExecutionListener implements JobExecutionListener, Ordered {
+  private final ContextStore<JobExecution, ContextAndScope> executionContextStore;
+
+  public TracingJobExecutionListener(
+      ContextStore<JobExecution, ContextAndScope> executionContextStore) {
+    this.executionContextStore = executionContextStore;
+  }
+
+  @Override
+  public void beforeJob(JobExecution jobExecution) {
+    Context context = tracer().startSpan(jobExecution);
+    Scope scope = context.makeCurrent();
```

Comment:
is before/after always synchronous? if so, a comment about that will help future readers not worry about context leaking here

Reply:
I'm 99.99% sure that it's always synchronous -- I'll add the comment.

And FYI: right now I'm testing all major spring batch features (wait for PRs 😄 ) so I'm definitely going to cover all multithreading/async scenarios.